### ISSUE TITLE
HttpRequestBuilder.add(name, value) behaves differently from Request.header(name, value)

### ISSUE THREAD

**Issue by ohadgur** [line/armeria#3932]
In case more than one value for headers is needed (for example, with `accept-encoding`), the 2 methods have different behaviors.

``` 
final HttpRequest request1 = HttpRequest.builder()
            .get("/")
            .header(HttpHeaderNames.ACCEPT_ENCODING, "gzip")
            .header(HttpHeaderNames.ACCEPT_ENCODING, "deflate")
            .header(HttpHeaderNames.ACCEPT_ENCODING, "gzip")
            .build();

System.out.println(request1.headers()); // [:method=GET, :path=/, accept-encoding=gzip] 


HttpRequest request2 = HttpRequest.builder().get("/")
            .build();
request2 = request2.withHeaders(request2.headers().toBuilder()
            .add(HttpHeaderNames.ACCEPT_ENCODING, "gzip")
            .add(HttpHeaderNames.ACCEPT_ENCODING, "deflate")
            .add(HttpHeaderNames.ACCEPT_ENCODING, "gzip"));

        System.out.println(request2.headers()); // [:method=GET, :path=/, accept-encoding=gzip, accept-encoding=deflate, accept-encoding=gzip]
```
IMO, they should behave the same (like `request2`)

In this particular case (duplicated gzip), this can cause issues in `DecodingClient` for example, since it expects the values of `accept-encoding` to be unique - if needed, I'll open a different issue for that.

**Comment by ikhoon** [comment]
As a workaround, we might want to use:
https://github.com/line/armeria/blob/5b384fbe27e7e6f9225d6db91cbb684d09dfbb5e/core/src/main/java/com/linecorp/armeria/common/AbstractHttpRequestBuilder.java#L222-L237

**Comment by ikhoon** [comment]
We still need to fix the behavior of `AbstractHttpRequestBuilder.header()`.

**Comment by icepeppermint** [comment]
@ikhoon, I'm not sure if it's the correct code, but I tried to fix it simply. (#3954)

**Comment by ikhoon** [comment]
Fixed by #3941.
